pkg/registry/apis/iam/hooks.go

```diff
@@ -168,3 +170,176 @@ func (b *IdentityAccessManagementAPIBuilder) AfterResourcePermissionCreate(obj r
+	// Convert IAM permissions to zanzana.RolePermission format
+	rolePerms := make([]zanzana.RolePermission, 0, len(permissions))
+	for _, perm := range permissions {
+		// Split the scope to get kind, attribute, identifier
+		kind, _, identifier := accesscontrol.SplitScope(perm.Scope)
+		rolePerms = append(rolePerms, zanzana.RolePermission{
+			Action:     perm.Action,
+			Kind:       kind,
+			Identifier: identifier,
+		})
+	}
+
+	// Translate to Zanzana tuples
+	openfgaTuples, err := zanzana.ConvertRolePermissionsToTuples(roleUID, rolePerms)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert directly to v1 tuples using common utility
+	v1Tuples := common.ToAuthzExtTupleKeys(openfgaTuples)
+
+	return v1Tuples, nil
+}
+
+// AfterCoreRoleCreate is a post-create hook that writes the core role permissions to Zanzana (openFGA)
+func (b *IdentityAccessManagementAPIBuilder) AfterCoreRoleCreate(obj runtime.Object, _ *metav1.CreateOptions) {
+	if b.zClient == nil {
+		return
+	}
+
+	wait := time.Now()
+	b.zTickets <- true
+	hooksWaitHistogram.Observe(time.Since(wait).Seconds())
+
+	role, ok := obj.(*iamv0.CoreRole)
+	if !ok {
+		return
+	}
+
+	go func(role *iamv0.CoreRole) {
+		defer func() {
+			<-b.zTickets
+		}()
+
+		roleUID := role.Name
+		permissions := role.Spec.Permissions
+
+		tuples, err := convertRolePermissionsToTuples(roleUID, permissions)
+		if err != nil {
+			b.logger.Error("failed to convert core role permissions to tuples",
+				"namespace", role.Namespace,
+				"roleUID", roleUID,
+				"err", err,
+				"permissionsCnt", len(permissions),
+			)
+			return
+		}
+
+		// Avoid writing if there are no valid tuples
+		if len(tuples) == 0 {
+			b.logger.Warn("no valid tuples to write for core role",
+				"namespace", role.Namespace,
+				"roleUID", roleUID,
+				"permissionsCnt", len(permissions),
+			)
+			return
+		}
+
+		b.logger.Debug("writing core role permissions to zanzana",
+			"namespace", role.Namespace,
+			"roleUID", roleUID,
+			"tuplesCnt", len(tuples),
+			"permissionsCnt", len(permissions),
+		)
+
+		ctx, cancel := context.WithTimeout(context.Background(), defaultWriteTimeout)
+		defer cancel()
+
+		err = b.zClient.Write(ctx, &v1.WriteRequest{
+			Namespace: role.Namespace,
+			Writes: &v1.WriteRequestWrites{
+				TupleKeys: tuples,
+			},
+		})
+		if err != nil {
+			b.logger.Error("failed to write core role permissions to zanzana",
+				"err", err,
+				"namespace", role.Namespace,
+				"roleUID", roleUID,
+				"tuplesCnt", len(tuples),
+			)
+		}
+	}(role.DeepCopy())
+}
+
+// AfterRoleCreate is a post-create hook that writes the role permissions to Zanzana (openFGA)
+func (b *IdentityAccessManagementAPIBuilder) AfterRoleCreate(obj runtime.Object, _ *metav1.CreateOptions) {
+	if b.zClient == nil {
+		return
+	}
+
+	wait := time.Now()
+	b.zTickets <- true
+	hooksWaitHistogram.Observe(time.Since(wait).Seconds())
+
+	role, ok := obj.(*iamv0.Role)
+	if !ok {
+		return
+	}
```

Comment:
Same as above:
```suggestion
	role, ok := obj.(*iamv0.Role)
	if !ok {
		return
	}

	wait := time.Now()
	b.zTickets <- true
	hooksWaitHistogram.Observe(time.Since(wait).Seconds())

```